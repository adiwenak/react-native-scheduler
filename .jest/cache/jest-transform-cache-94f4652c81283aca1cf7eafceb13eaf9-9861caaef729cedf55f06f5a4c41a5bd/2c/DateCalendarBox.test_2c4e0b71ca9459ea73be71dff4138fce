62eca380d1bdfec6be6904036236b6f5
var _interopRequireWildcard = require("@babel/runtime/helpers/interopRequireWildcard");

var _enzyme = require("enzyme");

var React = _interopRequireWildcard(require("react"));

var _reactTestRenderer = require("react-test-renderer");

var _data = require("../__stubsData/data");

var _DateCalendarBox = require("./DateCalendarBox");

describe("<DateCalendarBox />", function () {
  var onPressHandlerMock = jest.fn();
  it("should render with given props", function () {
    var component = (0, _reactTestRenderer.create)(React.createElement(_DateCalendarBox.DateCalendarBox, {
      date: 1,
      isSelected: true,
      isWeekend: false,
      whosBusy: _data.userCollections,
      dateBoxOnPressHandler: onPressHandlerMock,
      boxWidth: 50,
      dateFontSize: 16
    })).toJSON();
    expect(component).toMatchSnapshot();
  });
  it("should call onPressHandler when component is pressed", function () {
    var component = (0, _enzyme.shallow)(React.createElement(_DateCalendarBox.DateCalendarBox, {
      date: 1,
      isSelected: true,
      isWeekend: false,
      whosBusy: _data.userCollections,
      dateBoxOnPressHandler: onPressHandlerMock,
      boxWidth: 50,
      dateFontSize: 16
    }));
    component.props().onPress();
    expect(onPressHandlerMock.mock.calls.length).toBe(1);
  });
});
//# sourceMappingURL=data:application/json;charset=utf-8;base64,eyJ2ZXJzaW9uIjozLCJzb3VyY2VzIjpbIi9Vc2Vycy9ha3VybmFkaS9Qcm9qZWN0cy9yZWFjdC1uYXRpdmUtc2NoZWR1bGVyL2xpYnJhcnkvQ2FsZW5kYXJNb250aC9EYXRlQ2FsZW5kYXJCb3gvRGF0ZUNhbGVuZGFyQm94LnRlc3QudHN4Il0sIm5hbWVzIjpbXSwibWFwcGluZ3MiOiI7O0FBQUE7O0FBQ0E7O0FBQ0E7O0FBQ0E7O0FBQ0E7O0FBRUEsUUFBUSxDQUFDLHFCQUFELEVBQXdCLFlBQUE7QUFDNUIsTUFBTSxrQkFBa0IsR0FBRyxJQUFJLENBQUMsRUFBTCxFQUEzQjtBQUVBLEVBQUEsRUFBRSxDQUFDLGdDQUFELEVBQW1DLFlBQUE7QUFDakMsUUFBTSxTQUFTLEdBQUcsK0JBQ2Qsb0JBQUMsZ0NBQUQ7QUFDSSxNQUFBLElBQUksRUFBRSxDQURWO0FBRUksTUFBQSxVQUFVLEVBQUUsSUFGaEI7QUFHSSxNQUFBLFNBQVMsRUFBRSxLQUhmO0FBSUksTUFBQSxRQUFRLEVBQUUscUJBSmQ7QUFLSSxNQUFBLHFCQUFxQixFQUFFLGtCQUwzQjtBQU1JLE1BQUEsUUFBUSxFQUFFLEVBTmQ7QUFPSSxNQUFBLFlBQVksRUFBRTtBQVBsQixNQURjLEVBU1YsTUFUVSxFQUFsQjtBQVVBLElBQUEsTUFBTSxDQUFDLFNBQUQsQ0FBTixDQUFrQixlQUFsQjtBQUNILEdBWkMsQ0FBRjtBQWNBLEVBQUEsRUFBRSxDQUFDLHNEQUFELEVBQXlELFlBQUE7QUFFdkQsUUFBTSxTQUFTLEdBQUcscUJBQ2Qsb0JBQUMsZ0NBQUQ7QUFDSSxNQUFBLElBQUksRUFBRSxDQURWO0FBRUksTUFBQSxVQUFVLEVBQUUsSUFGaEI7QUFHSSxNQUFBLFNBQVMsRUFBRSxLQUhmO0FBSUksTUFBQSxRQUFRLEVBQUUscUJBSmQ7QUFLSSxNQUFBLHFCQUFxQixFQUFFLGtCQUwzQjtBQU1JLE1BQUEsUUFBUSxFQUFFLEVBTmQ7QUFPSSxNQUFBLFlBQVksRUFBRTtBQVBsQixNQURjLENBQWxCO0FBWUEsSUFBQSxTQUFTLENBQUMsS0FBVixHQUFrQixPQUFsQjtBQUNBLElBQUEsTUFBTSxDQUFDLGtCQUFrQixDQUFDLElBQW5CLENBQXdCLEtBQXhCLENBQThCLE1BQS9CLENBQU4sQ0FBNkMsSUFBN0MsQ0FBa0QsQ0FBbEQ7QUFDSCxHQWhCQyxDQUFGO0FBaUJILENBbENPLENBQVIiLCJzb3VyY2VzQ29udGVudCI6WyJpbXBvcnQgeyBzaGFsbG93IH0gZnJvbSBcImVuenltZVwiXG5pbXBvcnQgKiBhcyBSZWFjdCBmcm9tIFwicmVhY3RcIlxuaW1wb3J0IHsgY3JlYXRlIH0gZnJvbSBcInJlYWN0LXRlc3QtcmVuZGVyZXJcIlxuaW1wb3J0IHsgdXNlckNvbGxlY3Rpb25zIH0gZnJvbSBcIi4uL19fc3R1YnNEYXRhL2RhdGFcIlxuaW1wb3J0IHsgRGF0ZUNhbGVuZGFyQm94IH0gZnJvbSBcIi4vRGF0ZUNhbGVuZGFyQm94XCJcblxuZGVzY3JpYmUoXCI8RGF0ZUNhbGVuZGFyQm94IC8+XCIsICgpID0+IHtcbiAgICBjb25zdCBvblByZXNzSGFuZGxlck1vY2sgPSBqZXN0LmZuKClcblxuICAgIGl0KFwic2hvdWxkIHJlbmRlciB3aXRoIGdpdmVuIHByb3BzXCIsICgpID0+IHtcbiAgICAgICAgY29uc3QgY29tcG9uZW50ID0gY3JlYXRlKFxuICAgICAgICAgICAgPERhdGVDYWxlbmRhckJveFxuICAgICAgICAgICAgICAgIGRhdGU9ezF9XG4gICAgICAgICAgICAgICAgaXNTZWxlY3RlZD17dHJ1ZX1cbiAgICAgICAgICAgICAgICBpc1dlZWtlbmQ9e2ZhbHNlfVxuICAgICAgICAgICAgICAgIHdob3NCdXN5PXt1c2VyQ29sbGVjdGlvbnN9XG4gICAgICAgICAgICAgICAgZGF0ZUJveE9uUHJlc3NIYW5kbGVyPXtvblByZXNzSGFuZGxlck1vY2t9XG4gICAgICAgICAgICAgICAgYm94V2lkdGg9ezUwfVxuICAgICAgICAgICAgICAgIGRhdGVGb250U2l6ZT17MTZ9XG4gICAgICAgICAgICAvPikudG9KU09OKClcbiAgICAgICAgZXhwZWN0KGNvbXBvbmVudCkudG9NYXRjaFNuYXBzaG90KClcbiAgICB9KVxuXG4gICAgaXQoXCJzaG91bGQgY2FsbCBvblByZXNzSGFuZGxlciB3aGVuIGNvbXBvbmVudCBpcyBwcmVzc2VkXCIsICgpID0+IHtcblxuICAgICAgICBjb25zdCBjb21wb25lbnQgPSBzaGFsbG93KFxuICAgICAgICAgICAgPERhdGVDYWxlbmRhckJveFxuICAgICAgICAgICAgICAgIGRhdGU9ezF9XG4gICAgICAgICAgICAgICAgaXNTZWxlY3RlZD17dHJ1ZX1cbiAgICAgICAgICAgICAgICBpc1dlZWtlbmQ9e2ZhbHNlfVxuICAgICAgICAgICAgICAgIHdob3NCdXN5PXt1c2VyQ29sbGVjdGlvbnN9XG4gICAgICAgICAgICAgICAgZGF0ZUJveE9uUHJlc3NIYW5kbGVyPXtvblByZXNzSGFuZGxlck1vY2t9XG4gICAgICAgICAgICAgICAgYm94V2lkdGg9ezUwfVxuICAgICAgICAgICAgICAgIGRhdGVGb250U2l6ZT17MTZ9XG4gICAgICAgICAgICAvPlxuICAgICAgICApXG5cbiAgICAgICAgY29tcG9uZW50LnByb3BzKCkub25QcmVzcygpXG4gICAgICAgIGV4cGVjdChvblByZXNzSGFuZGxlck1vY2subW9jay5jYWxscy5sZW5ndGgpLnRvQmUoMSlcbiAgICB9KVxufSlcbiJdfQ==