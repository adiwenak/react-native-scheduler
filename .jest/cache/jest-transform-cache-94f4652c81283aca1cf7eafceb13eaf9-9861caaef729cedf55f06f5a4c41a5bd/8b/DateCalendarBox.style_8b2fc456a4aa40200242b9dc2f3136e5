fd1f5292e0c3fb63721c4895e3448dde
Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;

var _reactNative = require("react-native");

var _default = _reactNative.StyleSheet.create({
  touchableContainer: {
    flexDirection: "column"
  }
});

exports.default = _default;
//# sourceMappingURL=data:application/json;charset=utf-8;base64,eyJ2ZXJzaW9uIjozLCJzb3VyY2VzIjpbIi9Vc2Vycy9ha3VybmFkaS9Qcm9qZWN0cy9yZWFjdC1uYXRpdmUtc2NoZWR1bGVyL2xpYnJhcnkvQ2FsZW5kYXJNb250aC9EYXRlQ2FsZW5kYXJCb3gvRGF0ZUNhbGVuZGFyQm94LnN0eWxlLnRzIl0sIm5hbWVzIjpbXSwibWFwcGluZ3MiOiI7Ozs7O0FBQUE7O2VBTWUsd0JBQVcsTUFBWCxDQUF5QjtBQUNwQyxFQUFBLGtCQUFrQixFQUFFO0FBQ2hCLElBQUEsYUFBYSxFQUFFO0FBREM7QUFEZ0IsQ0FBekIsQyIsInNvdXJjZXNDb250ZW50IjpbImltcG9ydCB7IFN0eWxlU2hlZXQsIFZpZXdTdHlsZSB9IGZyb20gXCJyZWFjdC1uYXRpdmVcIlxuXG5pbnRlcmZhY2UgU3R5bGUge1xuICAgIHRvdWNoYWJsZUNvbnRhaW5lcjogVmlld1N0eWxlXG59XG5cbmV4cG9ydCBkZWZhdWx0IFN0eWxlU2hlZXQuY3JlYXRlPFN0eWxlPih7XG4gICAgdG91Y2hhYmxlQ29udGFpbmVyOiB7XG4gICAgICAgIGZsZXhEaXJlY3Rpb246IFwiY29sdW1uXCJcbiAgICB9XG59KVxuIl19