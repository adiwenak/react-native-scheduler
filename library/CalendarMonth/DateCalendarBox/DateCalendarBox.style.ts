import { StyleSheet, ViewStyle } from "react-native"

interface Style {
    touchableContainer: ViewStyle
}

export default StyleSheet.create<Style>({
    touchableContainer: {
        flexDirection: "column"
    }
})
