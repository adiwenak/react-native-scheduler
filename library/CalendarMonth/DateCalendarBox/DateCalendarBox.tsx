import * as React from "react"
import { Component } from "react"
import { TouchableOpacity } from "react-native"
import { DateNumber, User } from "../../shared/model"
import styles from "./DateCalendarBox.style"
import { BusyIndicator, DateComponent } from "./IndividualComponents"

interface ComponentProps {
  date?: DateNumber
  isSelected: boolean
  isWeekend: boolean
  boxWidth: number
  dateFontSize: number
  whosBusy: User[]
  dateBoxOnPressHandler: (date: DateNumber) => void
}

export class DateCalendarBox extends Component<ComponentProps> {
  public handleOnPress = () => {
    if (this.props.date) {
      this.props.dateBoxOnPressHandler(this.props.date)
    }
  }

  public render() {
    const dateComponentHeight = this.props.boxWidth * 0.6
    const busyIndicatorHeight = this.props.boxWidth * 0.2
    return (
      <TouchableOpacity
        style={[
          styles.touchableContainer,
          { height: this.props.boxWidth * 0.8, width: this.props.boxWidth }]}
        onPress={this.handleOnPress}
      >
        <DateComponent
          date={this.props.date}
          isWeekend={this.props.isWeekend}
          isSelected={this.props.isSelected}
          dateFontSize={this.props.dateFontSize}
          dateComponentHeight={dateComponentHeight}
        />
        <BusyIndicator whosBusy={this.props.whosBusy} height={busyIndicatorHeight} />
      </TouchableOpacity>
    )
  }
}
