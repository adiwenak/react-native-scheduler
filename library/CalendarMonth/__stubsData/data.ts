import moment from "moment"
import { Colours } from "../../shared/colour"
import { Activity, Category, CategoryActivities, CategorySelection } from "../../shared/model"
import { CategoryActivity } from "../CalendarMonth"

export const cat1: CategorySelection = {
  id: "1",
  name: "Ben Af",
  colour: Colours.Cobalt,
  isSelected: false
}

export const cat2: CategorySelection = {
  id: "2",
  name: "Cardano",
  colour: Colours.Cardinal,
  isSelected: false
}

export const cat3: CategorySelection = {
  id: "3",
  name: "Zester",
  colour: Colours.Zest,
  isSelected: false
}

export const cat4: CategorySelection = {
  id: "4",
  name: "Green Lantern",
  colour: Colours.LaPalma,
  isSelected: false
}

export const categoryCollections: Category[] = [cat1, cat2, cat3, cat4]

const date1 = moment().add(1, "days")
const date2 = moment().add(2, "days")
const date3 = moment().add(3, "days")
const date4 = moment().add(4, "days")
const date5 = moment().add(1, "weeks")
const date6 = moment().add(1, "weeks").subtract(1, "days")

export const allActivities: Activity[] = [
  {
    id: "1-1",
    category: "1",
    name: "Clean Toilet",
    startTime: date1.toDate(),
    endTime: date1.add(2, "hours").toDate()
  },
  {
    id: "1-2",
    category: "1",
    name: "Gardening",
    startTime: date2.add(1, "hours").toDate(),
    endTime: date2.add(3, "hours").toDate()
  },
  {
    id: "1-3",
    category: "1",
    name: "Gardening",
    startTime: date6.toDate(),
    endTime: date6.add(4, "hours").toDate()
  },
  {
    id: "2-1",
    category: "2",
    name: "Clean Toilet",
    startTime: date1.add(3, "hours").toDate(),
    endTime: date1.add(5, "hours").toDate()
  },
  {
    id: "2-2",
    category: "2",
    name: "Gardening",
    startTime: date2.add(2, "hours").toDate(),
    endTime: date2.add(4, "hours").toDate()
  },
  {
    id: "3-1",
    name: "Clean Toilet",
    category: "3",
    startTime: date3.add(1, "hours").toDate(),
    endTime: date3.add(5, "hours").toDate()
  },
  {
    id: "3-2",
    category: "3",
    name: "Gardening",
    startTime: date4.add(2, "hours").toDate(),
    endTime: date4.add(4, "hours").toDate()
  },
  {
    id: "4-1",
    category: "4",
    name: "BBQ",
    startTime: date1.add(4, "hours").toDate(),
    endTime: date1.add(6, "hours").toDate()
  },
  {
    id: "4-2",
    category: "4",
    name: "Clean Toilet",
    startTime: date5.add(1, "hours").toDate(),
    endTime: date5.add(5, "hours").toDate()
  },
  {
    id: "4-3",
    category: "4",
    name: "Gardening",
    startTime: date6.toDate(),
    endTime: date6.add(4, "hours").toDate()
  }
]

const map: CategoryActivities = {}

for (const activity of allActivities) {
  if (!map[activity.category]) {
    map[activity.category] = []
  }

  map[activity.category].push(activity)
}

const categoryActivities: CategoryActivity[] = []

Object.keys(map).forEach((categoryId: string) => {
  const category = categoryCollections.find((cat) => cat.id === categoryId)
  if (!category) {
    return
  }

  categoryActivities.push({
    category,
    tasks: map[categoryId]
  })
})

export { categoryActivities }
