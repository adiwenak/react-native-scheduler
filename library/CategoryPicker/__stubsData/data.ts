import { Category, CategorySelection, Colours } from "../../shared"

export const cat1: CategorySelection = {
    id: "1",
    name: "Ben Af",
    colour: Colours.Cobalt,
    isSelected: false
}

export const cat2: CategorySelection = {
    id: "2",
    name: "Cardano",
    colour: Colours.Cardinal,
    isSelected: false
}

export const cat3: CategorySelection = {
    id: "3",
    name: "Zester",
    colour: Colours.Zest,
    isSelected: false
}

export const cat4: CategorySelection = {
    id: "4",
    name: "Green Lantern",
    colour: Colours.LaPalma,
    isSelected: false
}

export const userSelectionCollection: CategorySelection[] = [
    cat1,
    cat2,
    cat3,
    cat4
]
