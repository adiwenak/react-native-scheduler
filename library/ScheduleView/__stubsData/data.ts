import moment from "moment"
import { Activity } from "../../shared"
import { Colours } from "../../shared/colour"

const date1 = moment().add(1, "days").set({ hour: 0, minute: 0, second: 0, millisecond: 0 })

export const dataGroup1: Activity[] = [
  {
    id: "A1-2",
    category: "A1",
    startTime: date1.set({ hour: 13, minute: 0 }).toDate(),
    endTime: date1.set({ hour: 15, minute: 0 }).toDate(),
    name: "Test A1-2"
  },
  {
    id: "A2-1",
    category: "A2",
    startTime: date1.set({ hour: 13, minute: 30 }).toDate(),
    endTime: date1.set({ hour: 15, minute: 30 }).toDate(),
    name: "Test A2-1"
  },
  {
    id: "A1-1",
    category: "A1",
    startTime: date1.set({ hour: 11, minute: 30 }).toDate(),
    endTime: date1.set({ hour: 14, minute: 30 }).toDate(),
    name: "Test A1-1"
  },
  {
    id: "A2-2",
    category: "A2",
    startTime: date1.set({ hour: 14, minute: 0 }).toDate(),
    endTime: date1.set({ hour: 16, minute: 30 }).toDate(),
    name: "Test A2-2"
  }
]

export const dataGroup2: Activity[] = [
  {
    id: "B1-2",
    category: "B1",
    startTime: date1.set({ hour: 10, minute: 0 }).toDate(),
    endTime: date1.set({ hour: 11, minute: 0 }).toDate(),
    name: "Test B1-2"
  },
  {
    id: "B2-1",
    category: "B2",
    startTime: date1.set({ hour: 9, minute: 0 }).toDate(),
    endTime: date1.set({ hour: 11, minute: 0 }).toDate(),
    name: "Test B2-1"
  },
  {
    id: "B1-1",
    category: "B1",
    startTime: date1.set({ hour: 8, minute: 30 }).toDate(),
    endTime: date1.set({ hour: 10, minute: 30 }).toDate(),
    name: "Test B1-1"
  }
]

export const dataGroup3: Activity[] = [
  {
    id: "C1-1",
    category: "C1",
    startTime: date1.set({ hour: 11, minute: 10 }).toDate(),
    endTime: date1.set({ hour: 12, minute: 30 }).toDate(),
    name: "Test C1-1"
  },
  {
    id: "C1-2",
    category: "C1",
    startTime: date1.set({ hour: 12, minute: 20 }).toDate(),
    endTime: date1.set({ hour: 13, minute: 15 }).toDate(),
    name: "Test C1-2"
  }
]

export const dataGroup4: Activity[] = [
  {
    id: "D1-1",
    category: "D1",
    startTime: date1.set({ hour: 13, minute: 40 }).toDate(),
    endTime: date1.set({ hour: 15, minute: 0 }).toDate(),
    name: "Test D1-1"
  },
  {
    id: "D2-1",
    category: "D2",
    startTime: date1.set({ hour: 11, minute: 50 }).toDate(),
    endTime: date1.set({ hour: 14, minute: 40 }).toDate(),
    name: "Test D2"
  }
]

export const dataGroup5: Activity[] = [
  {
    id: "E1-1",
    category: "E1",
    startTime: date1.set({ hour: 15, minute: 10 }).toDate(),
    endTime: date1.set({ hour: 18, minute: 20 }).toDate(),
    name: "Test E1-1"
  },
  {
    id: "E2-1",
    category: "E2",
    startTime: date1.set({ hour: 16, minute: 35 }).toDate(),
    endTime: date1.set({ hour: 17, minute: 5 }).toDate(),
    name: "Test E2-1"
  },
  {
    id: "E2-2",
    category: "E2",
    startTime: date1.set({ hour: 14, minute: 45 }).toDate(),
    endTime: date1.set({ hour: 16, minute: 25 }).toDate(),
    name: "Test E2-2"
  }
]

export const groupBackgroundColourMap = {
  A1: Colours.Indigo,
  A2: Colours.LightBlue,
  B1: Colours.Purple,
  B2: Colours.DeepPurple,
  C1: Colours.Teal,
  C2: Colours.Lime,
  D1: Colours.Yellow,
  D2: Colours.Orange,
  E1: Colours.Cardinal,
  E2: Colours.Pink
}
