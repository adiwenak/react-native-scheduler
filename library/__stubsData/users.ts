import { UserSelection, User } from '../shared';
import { Colours } from '../shared/colour'

export const user1: UserSelection = {
    id: "1",
    name: "Ben Af",
    colourIndicator: Colours.Cobalt,
    isSelected: false
}

export const user2: UserSelection = {
    id: "2",
    name: "Cardano",
    colourIndicator: Colours.Cardinal,
    isSelected: false
}

export const user3: UserSelection = {
    id: "3",
    name: "Zester",
    colourIndicator: Colours.Zest,
    isSelected: false
}

export const user4: UserSelection = {
    id: "4",
    name: "Green Lantern",
    colourIndicator: Colours.LaPalma,
    isSelected: false
}

export const userSelectionCollection: UserSelection[] = [
    user1,
    user2,
    user3,
    user4
]

export const userCollections: User[] = [
    {id: "1", name: "Adam", colourIndicator: "blue"},
    {id: "2", name: "Adi", colourIndicator: "green"},
    {id: "3", name: "Bruno", colourIndicator: "red"},
    {id: "4", name: "George", colourIndicator: "pink"},
    {id: "5", name: "Chiro", colourIndicator: "black"},
    {id: "6", name: "Olivia", colourIndicator: "grey"}
]
