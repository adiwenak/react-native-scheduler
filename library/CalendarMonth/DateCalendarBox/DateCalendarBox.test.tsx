import { shallow } from "enzyme"
import * as React from "react"
import { create } from "react-test-renderer"
import { userCollections } from "../__stubsData/data"
import { DateCalendarBox } from "./DateCalendarBox"

describe("<DateCalendarBox />", () => {
    const onPressHandlerMock = jest.fn()

    it("should render with given props", () => {
        const component = create(
            <DateCalendarBox
                date={1}
                isSelected={true}
                isWeekend={false}
                whosBusy={userCollections}
                dateBoxOnPressHandler={onPressHandlerMock}
                boxWidth={50}
                dateFontSize={16}
            />).toJSON()
        expect(component).toMatchSnapshot()
    })

    it("should call onPressHandler when component is pressed", () => {

        const component = shallow(
            <DateCalendarBox
                date={1}
                isSelected={true}
                isWeekend={false}
                whosBusy={userCollections}
                dateBoxOnPressHandler={onPressHandlerMock}
                boxWidth={50}
                dateFontSize={16}
            />
        )

        component.props().onPress()
        expect(onPressHandlerMock.mock.calls.length).toBe(1)
    })
})
