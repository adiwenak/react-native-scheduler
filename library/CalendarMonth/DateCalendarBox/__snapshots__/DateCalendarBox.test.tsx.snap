// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<DateCalendarBox /> should render with given props 1`] = `
<View
  accessibilityComponentType={undefined}
  accessibilityLabel={undefined}
  accessibilityTraits={undefined}
  accessible={true}
  collapsable={undefined}
  hasTVPreferredFocus={undefined}
  hitSlop={undefined}
  isTVSelectable={true}
  nativeID={undefined}
  onLayout={undefined}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "flexDirection": "column",
      "height": 40,
      "opacity": 1,
      "width": 50,
    }
  }
  testID={undefined}
  tvParallaxProperties={undefined}
>
  <View
    style={
      Object {
        "alignItems": "center",
        "flex": 1,
        "justifyContent": "center",
      }
    }
  >
    <View
      style={
        Array [
          Object {
            "backgroundColor": "#FF1493",
          },
          Object {
            "borderRadius": 12.5,
            "height": 25,
            "width": 25,
          },
        ]
      }
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "flex": 1,
            "justifyContent": "center",
          }
        }
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Array [
              Object {
                "color": "#323A48",
                "fontSize": 14,
                "fontWeight": "500",
              },
              Object {
                "fontSize": 16,
              },
            ]
          }
        >
          1
        </Text>
      </View>
    </View>
  </View>
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "flex": 1,
          "flexDirection": "row",
          "justifyContent": "center",
        },
        Object {
          "height": 10,
          "paddingBottom": 8,
        },
      ]
    }
  >
    <View
      style={
        Object {
          "backgroundColor": "blue",
          "borderRadius": 2.9411764705882355,
          "height": 5.882352941176471,
          "marginLeft": 2,
          "marginRight": 2,
          "width": 5.882352941176471,
        }
      }
    />
    <View
      style={
        Object {
          "backgroundColor": "green",
          "borderRadius": 2.9411764705882355,
          "height": 5.882352941176471,
          "marginLeft": 2,
          "marginRight": 2,
          "width": 5.882352941176471,
        }
      }
    />
    <View
      style={
        Object {
          "backgroundColor": "red",
          "borderRadius": 2.9411764705882355,
          "height": 5.882352941176471,
          "marginLeft": 2,
          "marginRight": 2,
          "width": 5.882352941176471,
        }
      }
    />
    <View
      style={
        Object {
          "backgroundColor": "pink",
          "borderRadius": 2.9411764705882355,
          "height": 5.882352941176471,
          "marginLeft": 2,
          "marginRight": 2,
          "width": 5.882352941176471,
        }
      }
    />
    <View
      style={
        Object {
          "backgroundColor": "black",
          "borderRadius": 2.9411764705882355,
          "height": 5.882352941176471,
          "marginLeft": 2,
          "marginRight": 2,
          "width": 5.882352941176471,
        }
      }
    />
    <View
      style={
        Object {
          "backgroundColor": "grey",
          "borderRadius": 2.9411764705882355,
          "height": 5.882352941176471,
          "marginLeft": 2,
          "marginRight": 2,
          "width": 5.882352941176471,
        }
      }
    />
  </View>
</View>
`;
